neuralcompression/layers/_generalized_divisive_normalization.py

```diff
@@ -0,0 +1,231 @@
+            generalized divisive normalization; the division is replaced by
+            multiplication).
+        rectify: If ``True``, apply a ``ReLU`` non-linearity to the inputs
+            before calculating the generalized divisive normalization response.
+        alpha_parameter: A ``Tensor`` means that the value of ``alpha`` is
+            fixed. A ``Callable`` can be used to determine the value of
+            ``alpha`` as a function of some other parameter or tensor. This can
+            be a ``Parameter``. ``None`` means that when the layer is
+            initialized, a ``NonNegativeParameterization`` layer is created to
+            train ``alpha`` (with a minimum value of ``1``). The default is a
+            fixed value of ``1``.
+        beta_parameter: A ``Tensor`` means that the value of ``beta`` is fixed.
+            A ``Callable`` can be used to determine the value of ``beta`` as a
+            function of some other parameter or tensor. This can be a
+            ``Parameter``. ``None`` means that when the layer is initialized, a
+            ``NonNegativeParameterization`` layer is created to train ``beta``
+            (with a minimum value of ``1e-6``).
+        epsilon_parameter: A ``Tensor`` means that the value of ``epsilon`` is
+            fixed. A ``Callable`` can be used to determine the value of
+            ``epsilon`` as a function of some other parameter or tensor. This
+            can be a ``Parameter``. ``None`` means that when the layer is
+            initialized, a ``NonNegativeParameterization`` layer is created to
+            train ``epsilon`` (with a minimum value of 1e-6). The default is a
+            fixed value of ``1``.
+        gamma_parameter: A ``Tensor`` means that the value of ``gamma`` is
+            fixed. A ``Callable`` can be used to determine the value of
+            ``gamma`` as a function of some other parameter or tensor. This can
+            be a ``Parameter``. ``None`` means that when the layer is
+            initialized, a ``NonNegativeParameterization`` layer is created to
+            train ``gamma``.
+        alpha_initializer: initializes the ``alpha`` parameter. Only used if
+            ``alpha`` is trained. Defaults to ``1``.
+        beta_initializer: initializes the ``beta`` parameter. Only used if
+            ``beta`` is created when initializing the layer. Defaults to ``1``.
+        epsilon_initializer: initializes the ``epsilon`` parameter. Only used
+            if ``epsilon`` is trained. Defaults to ``1``.
+        gamma_initializer: initializes the ``gamma`` parameter. Only used if
+            ``gamma`` is created when initializing the layer. Defaults to the
+            identity multiplied by ``0.1``. A good default value for the
+            diagonal is somewhere between ``0`` and ``0.5``. If set to ``0``
+            and ``beta`` is initialized as ``1``, the layer is effectively
+            initialized to the identity operation.
```

Comment:
This API is too complicated. One thing in particular I would like to change is divergent behavior based on input type. The best option is for every parameter of the class to have one type. The second best option is to have union types where the types in the Union are functionally similar (think `Path` and `os.path_like`). A `Tensor` and a `Callable` are completely different though.

The reason it is too complicated is there are too many features. If you want all of these features, I would split them out into helper functions and document the spectrum of helper functions available to the user. Another way to reduce the API complication is to use a `NamedTuple` that clearly semantically spaces out what you are putting in as the parameters of the class.